### PR TITLE
adding testing for tic_tac_toe.rb module

### PR DESCRIPTION
**Pull request by eduardoreisalvarenga**
Testing our Ruby TicTacToe program using RSpec

### CODE REVIEW

**Comment by euqueme** [comment]
Hello, @eduardoreisalvarenga and @RichardBongani :wave: 

I can't start reviewing your project because you didn't provide a description in your pull request, please do it and request for a code review again

PROJECT STATUS: INVALID :no_good_woman:

**Comment by pelzolga123** [comment]
## Hi Team!  
### Good start :+1: 
Still there are things, that should be improved:
* [x] Linters config: Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))
* [x] Specs for each method should be wrapped in one `describe` block. As in example here https://github.com/rubocop-hq/rspec-style-guide#empty-lines-around-examples . ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/describe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
* [x] All the public methods of the classes implemented should be tested ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/Ruby_testing/all_tic_tac_toe.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))
* [x] You are asking user to play game for the tests to work, but it would be better if you created test cases yourself. All test should work when you run `rspec`, but when user is playing not all of them are called. So it would be better if you change this approach. 

### Please kindly fix the above issues and send another code review request.

If you have any questions or need help, you can message me on slack (@ Olga Pelts). *Happy coding*

File: spec/tic_tac_toe_spec.rb [hunk 1]
```diff
@@ -1,15 +1,38 @@
-# require "rspec/autorun"
-require_relative "../lib/tic_tac_toe.rb"
+require_relative '../bin/main.rb'
+require_relative '../lib/tic_tac_toe.rb'
+require_relative '../lib/player.rb'
```
Comment (pelzolga123) [inline spec/tic_tac_toe_spec.rb]: You don't have any tests for player class. It's a small class but you can test `initialize` method.

**Comment by Anna-Myzukina** [comment]
## Hi team !  

# Status: [changes required]


### Good job so far, just a couple more changes and your project can be approved!

#### Project specific


* [x] All the public methods of the classes implemented in the project should be tested screenshot ( from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/))

* [x] Specs should cover positive and negative scenarios screenshot (from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59566098/)) Example: there is a spec for correct usage of method \[e.g. `it "creates object if all required attributes are passed"`\] and also there is spec for wrong usage of method \[e.g. `it "does not create object if required attributes are missing"`\].



#### General

* Kindly recommended adding to Pull Request a short description 
 (read more in [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
screenshot

*  Comments: It’s better to delete commented code. If you need to review a previous version of your code, use git history screenshot (from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))
* This branch has conflicts that must be resolved
error


### Make changes and submit another code review request.

*If you have any questions, let me know on slack (@Anna-Muzykina) and I will be glad to clarify with you.*
*Happy coding * <g-emoji class="g-emoji" alias="computer" fallback-src="https://github.githubassets.com/images/icons/emoji/unicode/1f4bb.png">computer</g-emoji>

File: spec/tic_tac_toe_spec.rb (hunk 1 above)
Comment (eduardoreisalvarenga) [reply spec/tic_tac_toe_spec.rb]: Thanks for your comments! We tested the initialize.

**Review by MauricioRobayo — CHANGES_REQUESTED** [review]
Well done addressing the issues from the previous reviews!

It's getting there, but I'm facing an issue you might want to solve. When I run only the tests from `main_spec.rb` using `rspec spec/main_spec.rb`, I get `4 examples, 2 failures`:

ttt-tests

<details><summary>Click here for full output</summary>

$ rspec spec/main_spec.rb 


      0  |  1 |  2
      ----------------
      3  |  4 |  5
      ----------------
      6  |  7 |  8
.Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" 
Player 1 choose a valid symbol:
Player 1 choose a symbol "X" or "O" F.

          0  |  1 |  2
          ----------------
          3  |  4 |  5
          ----------------
          6  |  7 |  8
It's Player 1's turn
Pick a position between 0-8: F

Failures:

  1) Main checks if player one picked an invalid symbol
     Failure/Error: expect(show_board.players_info).to be_a(Object)
     
     NoMethodError:
       undefined method `chomp' for nil:NilClass
     # ./spec/main_spec.rb:22:in `block (2 levels) in <top (required)>'

  2) Main runs a loop to play the game
     Failure/Error: expect(show_board.start_playing).to be(nil)
     
     NoMethodError:
       undefined method `chomp' for nil:NilClass
     # ./spec/main_spec.rb:30:in `block (2 levels) in <top (required)>'

Finished in 0.00637 seconds (files took 0.19132 seconds to load)
4 examples, 2 failures

Failed examples:

rspec ./spec/main_spec.rb:21 # Main checks if player one picked an invalid symbol
rspec ./spec/main_spec.rb:29 # Main runs a loop to play the game
</details>

* [x] Please double-check and fix main_spec.rb to pass all tests.

Status: __Request changes__.

**Review by mosesogwo — APPROVED** [review]
**HI team, your project is approved!**

**Cheers:stars:**

### Status: Approved.

**Pull request merged**